src/ROUTES.ts

```diff
@@ -1,30 +1,28 @@
-import {ValueOf} from 'type-fest';
+import {IsEqual, ValueOf} from 'type-fest';
 import CONST from './CONST';
 
-/**
- * This is a file containing constants for all the routes we want to be able to go to
- */
+// This is a file containing constants for all the routes we want to be able to go to
 
 /**
  * Builds a URL with an encoded URI component for the `backTo` param which can be added to the end of URLs
  */
-function getUrlWithBackToParam(url: string, backTo?: string): string {
-    const backToParam = backTo ? `${url.includes('?') ? '&' : '?'}backTo=${encodeURIComponent(backTo)}` : '';
-    return url + backToParam;
+function getUrlWithBackToParam<TUrl extends string>(url: TUrl, backTo?: string): `${TUrl}` | `${TUrl}?backTo=${string}` | `${TUrl}&backTo=${string}` {
```

Comment:
Not exactly sure how one would accomplish this, but I feel like a utility type for urls would be helpful, such that you could provide:

- a base URL
- an object type representing a series of optional or required query params (which of course can be in any order)

And then the resultant type would be any string matching the pattern of a valid URL with the correct base and query params.

Reply:
I feel like it should be doable @roryabraham, but I believe it's not something that should be handled in this PR, for two reasons:
- Preferably JS logic should be untouched in TS migration PRs
- This PR is a high prio for SWM team, as we are working on the navigation refactor. This migration could speed up things 😄 

Same as [here](https://github.com/Expensify/App/pull/31543#discussion_r1406093463), would you mind creating an issue for this, so it could be worked on in a separate PR?

Reply:
just submitted a proposal for #32143
@roryabraham